src/inference/include/openvino/runtime/infer_request.hpp

```diff
@@ -79,28 +79,92 @@ class OPENVINO_RUNTIME_API InferRequest {
      */
     void set_tensor(const ov::Output<ov::Node>& port, const Tensor& tensor);
 
+    /**
+     * @brief Sets batch of tensors for input data to infer by input name
+     * Model input shall have batch dimension and number of 'tensors' shall match with batch size
+     * Current version supports set tensors to model inputs only. In case if `name` is associated
+     * with output (or any other non-input node) - exception will be thrown
+     *
+     * @param name Name of input tensor.
+     * @param tensors Input tensors for batched infer request. The type of each tensor must match the model
+     * input precision and size (except batch dimension). Total size of tensors shall match with input's size
+     */
+    void set_tensors(const std::string& name, const std::vector<Tensor>& tensors);
```

Comment:
@myshevts @vladimir-paramuzov will it be supported if a single tensor inside the std::vector will have a batch != 1?
From autobatch perspective it seems OK, but not sure how GPU can understand that it will receive 4 tensors with batch = 2 instead of 8 tensors with batch 1.

Reply:
Auto-Batching doesn't "rebatch" the networks, it operates only batch1 networks/requests

Reply:
ok, now there is a check that single tensor has always batch=1

Reply:
What if user uses this method for outputs by mistake? Maybe set_input_tensors?